lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb

```diff
@@ -226,10 +238,43 @@ def client_rb_content(chef_server_url, node_name)
             https_proxy #{convergence_options[:bootstrap_proxy].inspect}
           EOM
         end
+        if convergence_options[:policy_name]
+          content << <<-EOM
+            # Policyfile Settings:
+            use_policyfile true
+            policy_document_native_api true
+            policy_group "#{convergence_options[:policy_group]}"
+            policy_name "#{convergence_options[:policy_name]}"
+          EOM
+        end
+
         content.gsub!(/^\s+/, "")
         content << convergence_options[:chef_config] if convergence_options[:chef_config]
         content
       end
+
+      def validate_policy_if_enabled
+        raise "policy_name was given but no policy_group. Both are required" if (convergence_options[:policy_name] &&
+                                                                                 !convergence_options[:policy_group])
+        raise "policy_group was given but no policy_name. Both are required" if (convergence_options[:policy_group] &&
+                                                                                 !convergence_options[:policy_name])
+        if convergence_options[:policy_name] && convergence_options[:chef_config]
+          entry_in_chef_config = []
+          ["use_policyfile", "policy_document_native_api", "policy_group", "policy_name"].each do |opts|
+            entry_in_chef_config << opts if convergence_options[:chef_config].include?(opts)
+          end
+
+          if !entry_in_chef_config.empty?
```

Comment:
I don't know if we need this logic here asserting that users didn't supply those configs in `:chef_config`.  What if they want to do something super weird like have `use_policyfile` reference a node attribute to determine if policyfiles are used?  I think we should keep our validation simple for now and add more validation as we se a need for it.